wallet.js

```diff
@@ -0,0 +1,354 @@
+  constructor({ network, server, seed }) {
+    super();
+
+    hathorLib.network.setNetwork(network);
+    this.network = network;
+
+    this.state = Wallet.CLOSED;
+    this.serverInfo = null;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
+
+    this.server = server;
+    this.seed = seed;
+
+    this.passphrase = '';
+    this.pinCode = '123';
+    this.password = '123';
+  }
+
+  /**
+   * Called when loading the history of transactions.
+   * It is called every HTTP Request to get the history of a set of addresses.
+   * Usually, this is called multiple times.
+   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
+   *
+   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
+   **/
+  onAddressesLoaded(result) {
+    //console.log('onAddressesLoaded', result);
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    if (value) {
+      this.setState(Wallet.SYNCING);
+      hathorLib.wallet.loadAddressHistory(0, hathorLib.constants.GAP_LIMIT).then(() => {
+        this.setState(Wallet.READY);
+      }).catch((error) => {
+        throw error;
+      })
+    } else {
+      // CONNECTING or CLOSED?
+      this.serverInfo = null;
+      this.setState(Wallet.CONNECTING);
+    }
+  }
+
+  getAddressToUse() {
+    return hathorLib.wallet.getAddressToUse();
+  }
+
+  getCurrentAddress() {
+    return hathorLib.wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      this.onNewTx(wsData);
+    }
+  }
+
+  reloadData() {
+    console.log('reloadData');
+  }
+
+  getHTRBalance() {
+    const historyTransactions = this.getTxHistory();
+    return hathorLib.wallet.calculateBalance(Object.values(historyTransactions), Wallet.HTR_TOKEN.uid);
+  }
+
+  getTxHistory() {
+    const data = hathorLib.wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in data ? data['historyTransactions'] : {};
+    return historyTransactions;
+  }
+
+  setState(state) {
+    this.state = state;
+    this.emit('state', state);
+  }
+
+  onNewTx(wsData) {
+    const newTx = wsData.history;
+
+    // TODO we also have to update some wallet lib data? Lib should do it by itself
+    const walletData = hathorLib.wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+
+    let isNewTx = false;
+    if (newTx.tx_id in historyTransactions) {
+      isNewTx = true;
```

Comment:
If the tx is in `historyTransactions` it's marked as new? Shouldn't it be the opposite?

Reply:
You're right. Fixed. Thanks!